web/js/playground.js

```diff
@@ -12,24 +12,54 @@ import { default as UTILS } from 'https://centerfordigitalhumanities.github.io/r
 import PLAYGROUND from './config.js';
 import ToolsCatalog from './tools.js';
 
+const RECENTLY_USED_KEY = 'recentlyUsedTools';
+
 /**
-*  Landing behavior for tools.  This should populate the set of available tools to the DOM.
-*/
-// function initializeTools(config) {
-//     console.log("initializing tools");
-//     return new Promise((res) => {
-//         let setContainer = document.getElementById(config.id)
-//         Array.from(config.catalog).forEach(tool => {
-//             setContainer.innerHTML += UTILS.thumbnailGenerator(tool)
-//         })
-//         UTILS.broadcast(undefined, PLAYGROUND.EVENTS.LOADED, setContainer, {})
-//         /**
-//          * Really each render should be a promise and we should return a Promise.all() here of some kind.
-//          * That would only work if PlaygroundRender resulted in a Promise where we could return Promise.all(renderPromises).
-//          */
-//         setTimeout(res, 200) //A small hack to ensure all the HTML generated by processing the views enters the DOM before this says it has resolved.
-//     })
-// }
+ * Retrieve recently used tools from local storage.
+ */
+function getRecentlyUsedTools() {
+    const recentTools = localStorage.getItem(RECENTLY_USED_KEY);
+    return recentTools ? JSON.parse(recentTools) : [];
+}
+
+/** 
+ * Save recently used tools to local storage.
+ */
+function saveRecentlyUsedTools(recentTools) {
+    localStorage.setItem(RECENTLY_USED_KEY, JSON.stringify(recentTools));
+}
+
+/**
+ * Update recently used tools, move the clicked tool to the top.
+ */
+function updateRecentlyUsedTools(clickedTool) {
+    const recentTools = getRecentlyUsedTools();
+    const existingIndex = recentTools.findIndex(tool => tool.label.toLowerCase() === clickedTool.label.toLowerCase());
+
+    console.log('clicked tool:', clickedTool);
+    console.log('existing index of tool in recent list:', existingIndex);
+    // remove the tools if it already exists in the list
+    if (existingIndex !== -1) {
+        recentTools.splice(existingIndex, 1);
+    }
+
```

Comment:
You have almost implemented it but the functionality blocks opening the tool. Also only navPlace Object and T-PEN are getting the label on top. We need it for all the tools in tool page.